src/box/memtx_engine.c

```diff
@@ -1061,6 +1244,42 @@ memtx_engine_new(const char *snap_dirname, bool force_recovery,
 		return NULL;
 	}
 
+	assert(allocator != NULL);
+	/* Default allocator */
+	if(!strcmp(allocator, "small")) {
+		memtx->alloc = &memtx->small_alloc;
+		memtx_tuple_new = small_memtx_tuple_new;
+		memtx_tuple_delete = small_memtx_tuple_delete;
+		memtx_tuple_format_vtab.tuple_delete = small_memtx_tuple_delete;
+		memtx_tuple_format_vtab.tuple_new = small_memtx_tuple_new;
+		memtx_tuple_format_vtab.tuple_chunk_delete = small_memtx_tuple_chunk_delete;
+		memtx_tuple_format_vtab.tuple_chunk_new = small_memtx_tuple_chunk_new;
+		memtx_allocator_init = small_allocator_init;
+		memtx_allocator_destroy = small_allocator_destroy;
+		memtx_allocator_enter_delayed_free_mode = small_allocator_enter_delayed_free_mode;
+		memtx_allocator_leave_delayed_free_mode = small_allocator_leave_delayed_free_mode;
+		memtx_allocator_stats = small_allocator_stats;
+		memtx_slab_cache_check = small_slab_cache_check;
+	} else if (!strcmp(allocator, "system")) {
+		memtx->alloc = &memtx->system_alloc;
+		memtx_tuple_new = system_memtx_tuple_new;
+		memtx_tuple_delete = system_memtx_tuple_delete;
+		memtx_tuple_format_vtab.tuple_delete = system_memtx_tuple_delete;
+		memtx_tuple_format_vtab.tuple_new = system_memtx_tuple_new;
+		memtx_tuple_format_vtab.tuple_chunk_delete = system_memtx_tuple_chunk_delete;
+		memtx_tuple_format_vtab.tuple_chunk_new = system_memtx_tuple_chunk_new;
+		memtx_allocator_init = system_allocator_init;
+		memtx_allocator_destroy = system_allocator_destroy;
+		memtx_allocator_enter_delayed_free_mode = system_allocator_enter_delayed_free_mode;
+		memtx_allocator_leave_delayed_free_mode = system_allocator_leave_delayed_free_mode;
+		memtx_allocator_stats = system_allocator_stats;
+		memtx_slab_cache_check = system_slab_cache_check;
```

Comment:
If we're going to introduce other allocators later, there would be even more such lengthy assignments. I suggest adding a macro for it.